core/src/Revolution/Processors/Context/GetList.php

```diff
@@ -79,7 +84,28 @@ public function prepareQueryBeforeCount(xPDOQuery $c)
                 'key:NOT IN' => is_string($exclude) ? explode(',', $exclude) : $exclude,
             ]);
         }
-
+        /*
+            When this class is used to fetch data for a grid filter's store (combo),
+            limit results to only those contexts present in the current grid.
+        */
+        if ($this->isGridFilter) {
+            if ($userGroup = $this->getProperty('usergroup', false)) {
+                $c->innerJoin(
+                    modAccessContext::class,
+                    'modAccessContext',
+                    [
+                        '`modAccessContext`.`target` = `modContext`.`key`',
+                        '`modAccessContext`.`principal` = ' . (int)$userGroup,
+                        '`modAccessContext`.`principal_class` = ' . $this->modx->quote(modUserGroup::class)
+                    ]
+                );
+                if ($policy = $this->getProperty('policy', false)) {
+                    $c->where([
+                        '`modAccessContext`.`policy`' => (int)$policy
+                    ]);
+                }
+            }
+        }
```

Comment:
Have you considered to use an early return of `$c` instead of using a huge if statement block?

Reply:
There have been other PRs I've worked on where an early return of X was suggested, helping me to learn the benefit of doing so in certain situations. I don't know that it really helps here. To me, doing what I think you're alluding to ...
```
if (!$this->isGridFilter) {
    return $c;
}
if ($userGroup = ...) {
[...]
return $c;
}

```
adds another couple lines and puts that return in a place you might not at first spot when scanning the code. Since an early return doesn't short circuit the parsing of any code, I don't see the benefit. That said, if you feel strongly about it (or obviously if I'm just missing something) I certainly don't mind changing it.

Reply:
I don’t have a strong opinion on this, I was just wondering if you did consider the early return and you did.